pyproject.toml

```diff
@@ -25,26 +26,36 @@ classifiers = [
     "Operating System :: Unix",
     "Operating System :: MacOS",
 ]
-requires-python = ">=3.9"
-dependencies = [
-    "aws-cdk.aws-lambda-python-alpha",
-    "aws-cdk-lib>=2.48.0,<3.0.0",
-    "constructs>=10.0.0,<11.0.0",
-]
 
-[project.optional-dependencies]
-test = [
-    "moto",
-    "opensearch-py",
-    "pytest",
-]
+[tool.poetry.dependencies]
+python = ">=3.9,<4"
 
-dev = [
-    "black",
-    "pre-commit",
-    "ruff",
-    "sds_data_manager[test]"
-]
+[tool.poetry.group.cdk-install.dependencies]
+# Required for aws-cdk.aws-lambda-python-alpha
+aws-cdk-lib = ">=2.66.0,<3.0.0"
+#aws-cdk-lib = "2.48.0"
+"aws-cdk.aws-lambda-python-alpha" = "^2.66.0a0"
+constructs = ">=10.0.0,<11.0.0"
+
+[tool.poetry.group.dev.dependencies]
+black = "^23.1.0"
+ruff = "^0.0.253"
+
+[tool.poetry.group.tests.dependencies]
+# Installed by default, can be installed without these dependencies using `poetry install --without tests`
+pytest = "==6.2.5"
+moto = "^4.1.3"
+boto3 = "^1.26.78"
+
+[tool.poetry.group.lambda-dev]
+# Not installed by default, can be installed using `poetry install --with cdk`
+optional = true
+
+[tool.poetry.group.lambda-dev.dependencies]
+opensearch-py = "^2.1.1"
+requests = "^2.28.2"
+python-jose = "^3.3.0"
+boto3 = "^1.26.78"
```

Comment:
```suggestion
```

* `python-jose`: We aren't using it yet from what I can tell, so we should add that back in again later.
* `boto3`: It does come pre-installed on Lambda's runtimes so do we want to install it here again? I can see either way on this one, so I don't have a strong preference.